infra/src/main/kotlin/fi/fta/geoviite/infra/error/ClientException.kt

```diff
@@ -178,3 +178,14 @@ class IntegrationNotConfiguredException(type: Integration) : ClientException(
     cause = null,
     localizedMessageKey = "error.integration-not-configured.$type",
 )
+
+class DuplicateDesignNameException(
```

Comment:
Jotenkin karsastan kokonaan uuden tyypin tekemistä näin spesifille jutulle, mutta toisaalta tätä patternia oli käytetty jo muualla. Käytännössähän tätä itse poikkeustyyppiä ei käytetä missään, mutta clientin aiheuttamaksi virheeksi tämä vähän niinkuin on ainakin mun päässäni eniten ajateltavissa (joku jossain koittaa tehdä typeriä asioita), `ClientException`:ia ei voi instantioida eikä mikään olemassaolevista tyypeistä ollut tarkoitettu tähän -> uusi tyyppi.

Reply:
Mä oon tottunut käyttämään virhelista-enumia sellaisissa kielissä jotka eivät hyödynnä exceptioneita ja palauttavat virheet arvoina 🤔. Komppaan kyllä että näiden uniikkien virheiden luominen on tällä tavoin raskaampaa, jolloin koodari ei jaksa niitä yhtä herkästi lähteä muodostamaan -> useammin epäselvempiä virheitä.

Vaikea sanoa mikä olisi mielekäs ratkaisu 🤔